skops/hub_utils/tests/test_hf_hub.py

```diff
@@ -1,28 +1,76 @@
 import json
 import os
+import pickle
 import shutil
 import tempfile
 from pathlib import Path
 from uuid import uuid4
 
 import pytest
 from huggingface_hub import HfApi
+from sklearn.linear_model import LogisticRegression
 
 from skops.hub_utils import download, get_config, get_requirements, init, push
 from skops.hub_utils._hf_hub import _create_config, _validate_folder
 from skops.hub_utils.tests.common import HF_HUB_TOKEN
 from skops.utils.fixes import metadata
 
 
-def _get_cwd():
-    """Return the current working directory.
+@pytest.fixture(scope="session")
+def repo_path():
+    with tempfile.TemporaryDirectory(prefix="skops-test-sample-repo") as repo_path:
+        yield Path(repo_path)
 
-    Only works if we're using pytest.
-    """
-    return Path(os.getenv("PYTEST_CURRENT_TEST").split("::")[0]).parent
 
+@pytest.fixture
+def destination_path():
+    with tempfile.TemporaryDirectory(prefix="skops-test") as dir_path:
+        yield Path(dir_path)
+
+
+@pytest.fixture(scope="session")
+def model_pickle(repo_path):
+    # Create a simple pickle file for the purpose of testing
+    clf = LogisticRegression()
+    clf.fit([[0, 1], [1, 0]], [0, 1])
+    path = repo_path / "model.pickle"
+    if path.exists():
+        raise OSError(f"File {path} already exists")
```

Comment:
since we're not using a temp folder, happy to have this removed. Don't really mind if you want to keep it though.

Reply:
Agree, this is not necessary.